docs/src/operate/registration/mqtt_api.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Entity management REST APIs
+tags: [Child-Device, Registration]
+sidebar_position: 1
+description: Register child devices and services with %%te%% MQTT APIs
+---
+
+# MQTT APIs for Entity Management
+
+The entity management MQTT APIs of %%te%% allows MQTT clients to register and deregister entities (child devices and services).
+
+## Register entity
+
+A new entity can be registered by publishing the entity definition to the MQTT topic that contains the **entity topic identifier**.
+The payload must contain at least the `@type` of the entity: whether it is a `child-device` or `service`.
+
+**Example: Register a child device**
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/child01//' '{
+  "@type": "child-device"
+}'
+```
+
+**Example: Register a service of a device**
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main/service/nodered' '{
+  "@type": "service"
+}'
+```
+
+:::note
+The supported `@type` values are `device`, `child-device` and `service`.
+The `device` type is reserved for the main %%te%% device which is pre-registered when it is bootstrapped.
+:::
+
+Other supported (optional) fields in the registration payload include:
+
+- `@parent`: Topic ID of the parent entity.
+  Required for nested child devices or services where the parent cannot be derived from the topic.
+- `@id`: External ID for the entity.
+- `@health`: Topic ID of the health endpoint service of this entity.
+  Valid only for `child-device` entities.
+  By default, it is the `tedge-agent` service on that device.
+
+**Example: Register a nested child device**
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/child01//' '{
+  "@type": "child-device",
+  "@parent": "device/child0",
+  "@id": "XYZ001:child01"
+}'
+```
+
+**Example: Register a child device with an external id**
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/child01//' '{
+  "@type": "child-device",
+  "@id": "XYZ001:child01"
+}'
+```
+
+Any additional fields included in the payload are considered as initial twin data for that entity,
+and they are re-published to the corresponding twin topics.
+
+**Example: Register a child device with initial twin data**
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/child01//' '{
+  "@type": "child-device",
+  "name": "Child 01",
+  "type": "Raspberry Pi 4"
+}'
+```
+
+:::warning
+Using the `@` prefix for such twin data keys is discouraged as this prefix is reserved by %%te%%.
+:::
+
+### Auto Registration
```

Comment:
Auto-registration is orthogonal to MQTT and HTTP. Hence, I would create a third independent page.

Reply:
Yeah I agree. Even I felt that adding auto-registration here kinda disrupted the flow of this document. But I still ended up adding it here as it is still linked to registration and MQTT. A dedicated doc, that is linked from this page sounds fine.

Reply:
Resolved by 4ff49e7